Embedding sbert class

elliegarcia opened the pull request:
Created a wrapper class for the SBERT embeddings. Please especially look over the hyperparameters in the SentenceTransformerEmbedding class

tnakkaya committed:
Esteli and I created sbert embedding class

Commit:
tried to import sentence_tranformers but the package is not found

Commit:
updated the sbert embedding class, so it's now working.

Commit:
added a few more parameters to the sbert embedding wrapper class

samuelgoodall commented on experiments/embedding_methods/sbert_embedding.py:
```diff
@@ -0,0 +1,31 @@
+        # Sentences are encoded by calling model.encode()
+        embedding = self.model.encode(sentence)
+        return embedding
+    def return_hyper_params(self):
```
add docstrings documentation: 
https://numpydoc.readthedocs.io/en/latest/format.html
https://realpython.com/documenting-python-code/#docstring-formats

elliegarcia replied:
I added more notation to the sbert wrapper class, could you look over it?

samuelgoodall commented on experiments/embedding_methods/sbert_embedding.py:
```diff
@@ -0,0 +1,31 @@
+        self.model_name = model_name
+
+    def embed_sentence(self, sentence: str):
+        """
```
Use Numpy docstring style:
https://numpydoc.readthedocs.io/en/latest/format.html
example is in the glove embedding class

samuelgoodall replied:
make sure the model that we use is getting the correct input-> does it have to be uncased?

samuelgoodall replied:
well have to make sure we dont cutoff long sentences like ansgar mentioned in the meeting

elliegarcia replied:
As I understand the uncased model strips the casing of the words, I think we don't need to feed it uncased sentences.
Added the 512 cutoff.

samuelgoodall reviewed (changes requested):
Looks good just minor changes for documentation style

Commit:
added a max sequence length

Commit:
added better notation, such as parameter and hyperparameter notation

Commit:
changed the sbert model that we use to: all-mpnet-base-v2
limited max_sequence_length to 128 tokens (as stated in the model description)
converted all sentences to lowercase

samuelgoodall committed:
Merge remote-tracking branch 'origin/main' into embedding_sbert_class

# Conflicts:
#	experiments/evaluation.py
